kyo-direct/shared/src/main/scala/kyo/Validate.scala

```diff
@@ -47,97 +82,52 @@ private[kyo] object Validate:
-                       |  IO(1).now
-                       |}
-                       |
-                       |defer {
-                       |  innerOperation.now  // OK - composing effects
-                       |}""".stripMargin)}""".stripMargin
-                )
-
             case tree @ ValDef(_, _, _) if tree.show.startsWith("var ") =>
                 fail(
                     tree,
                     s"""${"`var`".yellow} declarations are not allowed inside a ${"`defer`".yellow} block.
                        |
                        |Mutable state can lead to unexpected behavior with effects. Instead, use proper state management tools:
                        |
-                       |• Var (kyo-prelude)
-                       |• Atomic* classes (kyo-core)
-                       |• TRef and derivatives (kyo-stm)""".stripMargin
-                )
-
-            case Return(_, _) =>
-                fail(
-                    tree,
-                    s"""${"`return`".yellow} statements are not allowed inside a ${"`defer`".yellow} block.
-                       |
-                       |Early returns can break effect sequencing. Instead:
-                       |${highlight("""
-                       |// Instead of return:
-                       |defer {
-                       |  if condition then
-                       |    return value    // NOT OK - early return
-                       |  IO(1).now
-                       |}
-                       |
-                       |// Use if expressions:
-                       |defer {
-                       |  if condition then value
-                       |  else IO(1).now    // OK - explicit flow
-                       |}""".stripMargin)}""".stripMargin
+                       |• Most common: Atomic* classes (kyo-core)
+                       |• With transactional behavior: TRef and derivatives (kyo-stm)
+                       |• Advanced pure state management: Var (kyo-prelude)
+                       """.stripMargin
                 )
 
             case tree @ ValDef(_, _, _) if tree.show.startsWith("lazy val ") =>
                 fail(
                     tree,
-                    s"""${"`lazy val`".yellow} declarations are not allowed inside a ${"`defer`".yellow} block.
-                       |
-                       |Lazy evaluation can interfere with effect sequencing. Instead:
-                       |${highlight("""
-                       |// Instead of lazy val:
-                       |defer {
-                       |  lazy val x = IO(1).now  // NOT OK - lazy
-                       |  x + 1
-                       |}
-                       |
-                       |// Use regular val:
-                       |defer {
-                       |  val x = IO(1).now       // OK - eager
-                       |  x + 1
-                       |}""".stripMargin)}""".stripMargin
-                )
-
-            case Lambda(_, body) if !pure(body) =>
```

Comment:
This pattern match is wrong and I couldn't make it work. I've decided to remove because `dotty-cps-async` already produces an error message for this.